### ISSUE TITLE
Recruitment

### ISSUE THREAD

**Issue by ItsNeil17** [WillofSteel-Devs/app#8]
- Must be able to access all units for recruitment
- Must be able to specify how many units when recruiting
- (Optional) Give feedback on how what units are currently being recruited

### Linked PRs
* gh-20
* gh-25

**Comment by CozmoKitten** [comment]
This will require a dropdown menu to function cleanly. I will work on implementing this step.

**Comment by CozmoKitten** [comment]
Here's the drop down class I made as part of the recruitment code. Will be implemented as a component once I complete the GUI for the feature.
```py 
class Dropdown(tkinter.OptionMenu):
    def __init__(self, parent, options=[], default=None, *args, **kwargs):
        self.selected_value = tkinter.StringVar(parent)
        
        if default:
            self.selected_value.set(default)
        else:
            self.selected_value.set(options[0] if options else '')
        
        super().__init__(
            parent, 
            self.selected_value, 
            *options, 
            *args, 
            **kwargs)```

**Comment by CozmoKitten** [comment]
Making progress! Just a few backend issues I've asked Neil about.

**Comment by ItsNeil17** [comment]
How'd this code even run? 😂 
![image](https://github.com/WillofSteel-Devs/app/assets/68216747/2476562b-5b5e-4494-89d2-b692eebb9f8d)

that's an error on the fresh pull

**Comment by ItsNeil17** [comment]
https://github.com/WillofSteel-Devs/app/blob/667a296182f72511db9a3e11675ce7f818ccecaf/gui/frames/recruitment.py#L40-L42

`self.troopselector.get_selection()` returns `PY_VAR0` (or VAR1, 2 etc) not a troop type. Hence the JSON errors

**Comment by ItsNeil17** [comment]
Backend issues fixed with commit 4366071
Linting (pyright) fixed with commit 1e5abed
